app/src/main/java/com/douglasbruce/tasky/core/data/AlarmSchedulerImpl.kt

```diff
@@ -0,0 +1,128 @@
+package com.douglasbruce.tasky.core.data
+
+import android.app.AlarmManager
+import android.app.PendingIntent
+import android.content.Context
+import android.content.Intent
+import android.os.Build
+import com.douglasbruce.tasky.AlarmReceiver
+import com.douglasbruce.tasky.core.data.database.dao.EventDao
+import com.douglasbruce.tasky.core.data.database.dao.ReminderDao
+import com.douglasbruce.tasky.core.data.database.dao.TaskDao
+import com.douglasbruce.tasky.core.domain.mapper.toAlarmItem
+import com.douglasbruce.tasky.core.domain.mapper.toEvent
+import com.douglasbruce.tasky.core.domain.mapper.toReminder
+import com.douglasbruce.tasky.core.domain.mapper.toTask
+import com.douglasbruce.tasky.core.domain.utils.AlarmScheduler
+import com.douglasbruce.tasky.core.model.AlarmItem
+import dagger.hilt.android.qualifiers.ApplicationContext
+import kotlinx.coroutines.async
+import kotlinx.coroutines.supervisorScope
+import java.time.LocalDateTime
+import java.time.ZoneId
+import javax.inject.Inject
+
+class AlarmSchedulerImpl @Inject constructor(
+    @ApplicationContext private val context: Context,
+    private val eventsDao: EventDao,
+    private val tasksDao: TaskDao,
+    private val remindersDao: ReminderDao,
+) : AlarmScheduler {
+
+    private val alarmManager = context.getSystemService(AlarmManager::class.java)
+
+    override fun schedule(item: AlarmItem) {
+        if (shouldNotScheduleAlarmItem(item)) {
```

Comment:
Nitpick, but I'm a fan of always making booleans positive by default (so here `!shouldScheduleAlarmItem` instead). That reads easier when they're negated, since you will never need to do a double negation in your mind